File: matmuldb.py
import random
import time
	
n = 64
m1 = [[0]*n]*n
m2 = [[0]*n]*n

for i in range(0,n):
        for j in range(0,n):
            m1[i][j] = round(random.uniform(0,10e18),5)

for i in range(0,n):
		for j in range(0,n):
			print(m1[i][j], end=" ")
		print("\n")
            
                 
resultant_matrix = [[0]*n]*n

start = time.time()
for i in range(0,n):
	for j in range(0,n):
		for k in range(0,n):
			resultant_matrix = m1[i][k]*m2[k][j]
end = time.time()
meet_portion_time = end-start
print("Meet Portion Time: ",meet_portion_time, end="\n")

File: matmuldb.cpp
#include<bits/stdc++.h>
using namespace std;

const int n = 512;

long currentTime()
{
    struct timespec time;
    timespec_get(&time, CLOCK_MONOTONIC);
    return time.tv_sec * 1000000000 + time.tv_nsec;
}

void matrix_mul(double m1[][n], double m2[][n], int n){
    double resultant_matrix[n][n];
    long start = currentTime();
    for(int i = 0; i<n; i++){
        for(int j = 0; j<n; j++){
            resultant_matrix[i][j] = 0;
            for(int k = 0; k<n; k++){
                resultant_matrix[i][j] = m1[i][k]*m2[k][j];
            }
        }
    }
    long end = currentTime();
    for(int i = 0;i<n; i++){
    	for(int j = 0; j<n; j++){
    		cout<<resultant_matrix[i][j]<<" ";
    	}
    	cout<<endl;
    }
    cout<<endl<< "Meat Portion for N = "<<n<<" Time taken(in ns)= " << end - start;
    return;
}

int main(){
    double m1[n][n];
    double m2[n][n];
    for(int i = 0; i<n; i++){
        for(int j = 0; j<n; j++){
            m1[i][j] = rand();
            m2[i][j] = rand();
        }
    }
    matrix_mul(m1,m2,n);
    return 0;
}
